macros/channel_group_query.sql

```diff
@@ -0,0 +1,94 @@
+   WHEN lower(trim(mkt_medium)) IN ('referral', 'app', 'link') THEN 'Referral'
+   WHEN lower(trim(mkt_source)) IN ('email', 'e-mail', 'e_mail', 'e mail') OR lower(trim(mkt_medium)) IN ('email', 'e-mail', 'e_mail', 'e mail') THEN 'Email'
+   WHEN lower(trim(mkt_medium)) = 'affiliate' THEN 'Affiliates'
+   WHEN lower(trim(mkt_medium)) = 'audio' THEN 'Audio'
+   WHEN lower(trim(mkt_source)) = 'sms' OR lower(trim(mkt_medium)) = 'sms' THEN 'SMS'
+   WHEN lower(trim(mkt_medium)) like '%push' OR REGEXP_CONTAINS(trim(mkt_medium), r'(?i).*(mobile|notification).*') OR lower(trim(mkt_source)) = 'firebase' THEN 'Mobile Push Notifications'
+   ELSE 'Unassigned'
+END
+{% endmacro %}
+
+{% macro default__channel_group_query() %}
+CASE
+   WHEN lower(trim(mkt_source)) = '(direct)' AND lower(trim(mkt_medium)) IN ('(not set)', '(none)') THEN 'Direct'
+   WHEN lower(trim(mkt_medium)) like '%cross-network%' THEN 'Cross-network'
+   WHEN REGEXP_LIKE(lower(trim(mkt_medium)), '^(.*cp.*|ppc|retargeting|paid.*)$') THEN
+      CASE
+         WHEN upper(source_category) = 'SOURCE_CATEGORY_SHOPPING'
+            OR REGEXP_LIKE(lower(trim(mkt_campaign)), '^(.*(([^a-df-z]|^)shop|shopping).*)$') THEN 'Paid Shopping'
+         WHEN upper(source_category) = 'SOURCE_CATEGORY_SEARCH' THEN 'Paid Search'
+         WHEN upper(source_category) = 'SOURCE_CATEGORY_SOCIAL' THEN 'Paid Social'
+         WHEN upper(source_category) = 'SOURCE_CATEGORY_VIDEO' THEN 'Paid Video'
+         ELSE 'Paid Other'
+      END
+   WHEN lower(trim(mkt_medium)) IN ('display', 'banner', 'expandable', 'intersitial', 'cpm') THEN 'Display'
+   WHEN upper(source_category) = 'SOURCE_CATEGORY_SHOPPING'
+      OR REGEXP_LIKE(lower(trim(mkt_campaign)), '^(.*(([^a-df-z]|^)shop|shopping).*)$') THEN 'Organic Shopping'
+   WHEN upper(source_category) = 'SOURCE_CATEGORY_SOCIAL' OR lower(trim(mkt_medium)) IN ('social', 'social-network', 'sm', 'social network', 'social media') THEN 'Organic Social'
+   WHEN upper(source_category) = 'SOURCE_CATEGORY_VIDEO'
+      OR REGEXP_LIKE(lower(trim(mkt_medium)), '^(.*video.*)$') THEN 'Organic Video'
+   WHEN upper(source_category) = 'SOURCE_CATEGORY_SEARCH' OR lower(trim(mkt_medium)) = 'organic' THEN 'Organic Search'
+   WHEN lower(trim(mkt_medium)) IN ('referral', 'app', 'link') THEN 'Referral'
+   WHEN lower(trim(mkt_source)) IN ('email', 'e-mail', 'e_mail', 'e mail') OR lower(trim(mkt_medium)) IN ('email', 'e-mail', 'e_mail', 'e mail') THEN 'Email'
+   WHEN lower(trim(mkt_medium)) = 'affiliate' THEN 'Affiliates'
+   WHEN lower(trim(mkt_medium)) = 'audio' THEN 'Audio'
+   WHEN lower(trim(mkt_source)) = 'sms' OR lower(trim(mkt_medium)) = 'sms' THEN 'SMS'
+   WHEN lower(trim(mkt_medium)) like '%push' OR REGEXP_LIKE(lower(trim(mkt_medium)), '.*(mobile|notification).*') OR lower(trim(mkt_source)) = 'firebase' THEN 'Mobile Push Notifications'
+   ELSE 'Unassigned'
+END
+{% endmacro %}
+
+{% macro redshift__channel_group_query() %}
```

Comment:
Shouldn't this be `postgres__channel_group_query`? Or does postgres play nicely with the macro defined in `default__channel_group_query`?

Reply:
Correct, postgres will pick up default, but redshift has some differently named regexp functions